gst-plugin-pravega/src/pravegasink/imp.rs

```diff
@@ -562,6 +720,12 @@ impl BaseSinkImpl for PravegaSink {
             let buf_writer = BufWriter::with_capacity(settings.buffer_size, seekable_writer);
             let counting_writer = CountingWriter::new(buf_writer).unwrap();
 
+            let retention = Retention::new(settings.retention_type.clone(), settings.retention_days, settings.retention_bytes);
+            
+            let rentention_maintainer = RetentionMaintainer::new(retention, client_factory.clone(),
+                index_scoped_segment, scoped_segment);
+            rentention_maintainer.run();
```

Comment:
This thread needs to be stopped when the element is stopped. Make sure to join the thread before stop returns.

Reply:
done